src/status_im/browser/core.cljs

```diff
@@ -403,18 +403,34 @@
                                                   :error     %1
                                                   :result    %2}])]}))))
 
+(fx/defn handle-no-permissions [cofx {:keys [method id]} message-id]
+  (if (= method "eth_accounts")
+    ;; eth_accounts returns empty array for compatibility with meta-mask
+    (send-to-bridge cofx
+                    {:type      constants/web3-send-async-callback
+                     :messageId message-id
+                     :result    {:jsonrpc "2.0"
+                                 :id      (int id)
+                                 :result  []}})
+    (send-to-bridge cofx
+                    {:type      constants/web3-send-async-callback
+                     :messageId message-id
+                     :error     {:code 4100}})))
+
+(def require-permission?
```

Comment:
It is not conventional to add `?` suffix to a non-boolean value. `require-permission-enum` or `require-permission-set` is a more apt name.